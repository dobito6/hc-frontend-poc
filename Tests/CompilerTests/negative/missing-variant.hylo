//! no-std

struct A { public memberwise init }
struct B { public memberwise init }

fun f(x: auto A) auto -> B {
  let { (A(), B()) }
}
fun g(x: auto A) auto -> B {
  inout { B() }
}

public fun main() {
  var a = A()
  f(x: &a)
  g(x: a)
}
