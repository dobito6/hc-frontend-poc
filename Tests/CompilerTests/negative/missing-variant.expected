negative/missing-variant.hylo:15.3-11: error: cannot call bundle of type '[Void](x: auto A) auto -> B { let }' mutably
  f(x: &a)
  ~~~~~~~~
negative/missing-variant.hylo:16.3-10: error: cannot call bundle of type '[Void](x: auto A) auto -> B { inout }' immutably
  g(x: a)
  ~~~~~~~
